roms_tools/setup/cdr_forcing.py

```diff
@@ -886,180 +720,254 @@ def from_yaml(cls, filepath: Union[str, Path]) -> "CDRVolumePointSource":
-        list_allowed : bool, optional
-            If `True`, a list of release names is allowed. If `False`, only a single release name (string)
-            is allowed. Default is `True`.
+    Raises
+    ------
+    ValueError
+        If the number of valid releases exceeds the available colormap capacity.
 
-        Raises
-        ------
-        ValueError
-            If `releases` is not a string or list of strings, or if any release name is invalid (not in `self.releases`).
+    Notes
+    -----
+    The colormap is chosen dynamically based on the number of valid releases:
 
-        Notes
-        -----
-        This method checks that the `releases` input is in a valid format (either a string or a list of strings),
-        and ensures each release is present in the set of valid releases defined in `self.releases`. Invalid releases
-        are reported in the error message.
+    - If there are 10 or fewer releases, the "tab10" colormap is used.
+    - If there are more than 10 but fewer than or equal to 20 releases, the "tab20" colormap is used.
+    - For more than 20 releases, the "tab20b" colormap is used.
+    """
 
-        If `list_allowed` is set to `False`, only a single release name (string) will be accepted. Otherwise, a
-        list of release names is also acceptable.
-        """
+    # Determine the colormap based on the number of releases
+    if len(valid_releases) <= 10:
+        color_map = plt.get_cmap("tab10")
+    elif len(valid_releases) <= 20:
+        color_map = plt.get_cmap("tab20")
+    else:
+        color_map = plt.get_cmap("tab20b")
+
+    # Ensure the number of releases doesn't exceed the available colormap capacity
+    if len(valid_releases) > color_map.N:
+        raise ValueError(
+            f"Too many releases. The selected colormap supports up to {color_map.N} releases."
+        )
+
+    # Create a dictionary of colors based on the release indices
+    colors = {name: color_map(i) for i, name in enumerate(valid_releases)}
 
-        # Ensure that a list of releases is only allowed if `list_allowed` is True
-        if not list_allowed and not isinstance(releases, str):
+    return colors
+
+
+def _validate_release_location(grid, release: Release):
+    """Validates the closest grid location for a release site.
+
+    This function ensures that the given release site (lat, lon, depth) lies
+    within the ocean portion of the model grid domain. It:
+
+    - Checks if the point is within the grid domain (with buffer for boundary artifacts).
+    - Verifies that the location is not on land.
+    - Verifies that the location is not below the seafloor.
+    """
+    if grid:
+        # Adjust longitude based on whether it crosses the International Date Line (straddle case)
+        if grid.straddle:
+            lon = xr.where(release.lon > 180, release.lon - 360, release.lon)
+        else:
+            lon = xr.where(release.lon < 0, release.lon + 360, release.lon)
+
+        dx = 1 / grid.ds.pm
```

Comment:
any chance of divide by zero here?

Reply:
No... `grid.ds.pm` is always bigger than zero!